test/controllers/users_controller_test.rb

```diff
@@ -234,6 +240,15 @@ class UsersControllerTest < ActionController::TestCase
     assert_redirected_to '/users/extlogout'
   end
 
+  test "should redirect disabled external user to login page" do
+    Setting['authorize_login_delegation'] = true
+    Setting['authorize_login_delegation_auth_source_user_autocreate'] = 'apache'
+    users(:external_disabled).update(disabled: true)
```

Comment:
Fixtures would benefit more if the user is enabled by default, we change the disabled here anyway.
```suggestion
    users(:external).update(disabled: true)
```